library/src/main/scala/scala/scalajs/runtime/RuntimeString.scala

```diff
@@ -166,6 +166,32 @@ private[runtime] object RuntimeString {
     Pattern.matches(regex, thiz)
   }
 
+  // Both regionMatches ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/String.java
+  def regionMatches(thiz: String, ignoreCase: Boolean,
+                    toffset: Int, other: String,
+                    ooffset: Int, len: Int): Boolean = {
+    if (other == null) {
```

Comment:
Yes. If you invoke something on `thiz` immediately, this invocation will fail. However, in this method, we do other things first, so we could return a value, even if `thiz` is null. This violates evaluation order of the calling code (by how we hack the ops on top of the strings).